wdae/wdae/groups_api/tests/test_groups_rest.py

```diff
@@ -329,3 +330,52 @@ def test_cant_revoke_default_permissions(user_client, dataset):
 
         assert response.status_code == status.HTTP_403_FORBIDDEN
         assert dataset.groups.filter(pk=group.pk).exists()
+
+
+def test_groups_pagination(admin_client, hundred_groups):
+    url = "/api/v3/groups?page=1"
+    response = admin_client.get(url)
+    assert len(response.data) == 25
+    assert response.data[0]["name"] == "Group1"
+    assert response.data[24]["name"] == "Group30"
+    url = "/api/v3/groups?page=2"
+    response = admin_client.get(url)
+    assert len(response.data) == 25
+    assert response.data[0]["name"] == "Group31"
+    assert response.data[24]["name"] == "Group53"
+    url = "/api/v3/groups?page=3"
+    response = admin_client.get(url)
+    assert len(response.data) == 25
+    assert response.data[0]["name"] == "Group54"
+    assert response.data[24]["name"] == "Group76"
+    url = "/api/v3/groups?page=4"
+    response = admin_client.get(url)
+    assert len(response.data) == 25
+    assert response.data[0]["name"] == "Group77"
+    assert response.data[24]["name"] == "Group99"
+    url = "/api/v3/groups?page=5"
+    response = admin_client.get(url)
+    assert len(response.data) == 1
+    assert response.data[0]["name"] == "admin"
+
+def test_groups_search(admin_client, hundred_groups):
+    url = "/api/v3/groups?search=Group1"
+    response = admin_client.get(url)
+    assert len(response.data) == 12
+
+def test_groups_search_pagination(admin_client, hundred_groups):
+    url = "/api/v3/groups?page=1&search=Group"
```

Comment:
I know I am not a reviewer but I will allow myself the luxury of sticking my nose in:
This test can be made much clearer by using parameterizing it.

Reply:
Fixed